### PR TITLE
fixes #5572 clear unused exception for WScript.LoadScript

### LINKED ISSUE #5572: ChakraCore crashes with assertion failure in ScriptContext.cpp:2809
Debug build of ChakraCore received SIGILL with an ASSERTION FAILURE.

ASSERTION 15614: (/data/ChakraCore/lib/Runtime/Base/ScriptContext.cpp, line 2809) this->threadContext->GetRecordedException() == nullptr || GetThreadContext()->HasUnhandledException()
 Failure: (this->threadContext->GetRecordedException() == nullptr || GetThreadContext()->HasUnhandledException())

```
$ ./build.sh --debug
$ ./out/Debug/ch test.js
$ cat test.js
WScript.LoadScript('', '', {
    toString: function () {
      func_0();
    }
});
```

OS: Ubuntu 18.04 LTS
Arch: x86_64

### PR DESCRIPTION
**Pull request by duongnhn**
we try to convert inputs to string but for some invalid input, this call sets and records exception along with return error code to context thread. In this case, context thread need to clear unused exception before set and record new exception.

### CODE REVIEW

File: test/Bugs/bug_5572_wscript_loadscript_expection.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+  {
+    name: "Expect exception with invalid file name to WScript.LoadScript",
+    body: function () {  
+        assert.throws(function () { 
+            WScript.LoadScript(``, ``, {
```
Comment (kfarnung) [inline test/Bugs/bug_5572_wscript_loadscript_expection.js]: Nit, is there a reason you're using format strings here rather than just standard strings?

**Comment by jackhorton** [comment]
@dotnet-bot test all please

File: bin/ch/WScriptJsrt.cpp [hunk 2]
```diff
@@ -381,6 +382,14 @@ JsValueRef WScriptJsrt::LoadScriptHelper(JsValueRef callee, bool isConstructCall
 Error:
     if (errorCode != JsNoError)
     {
+        if (isInitializeThrow)
+        {
+            //clear unused exception
+            JsValueRef unusedException;
+            ChakraRTInterface::JsGetAndClearException(&unusedException);
```
Comment (kfarnung) [inline bin/ch/WScriptJsrt.cpp]: Could we just use `ChakraRTInterface::HasException` and clear based on that first?  That might be cleaner than having the macro set a local variable.

**Comment by akroshg** [comment]
There is LoadModule function as well. could you add a test to validate that it is working that as well?

File: test/Bugs/bug_5572_wscript_loadscript_expection.js (hunk 1 above)
Comment (duongnhn) [reply test/Bugs/bug_5572_wscript_loadscript_expection.js]: Absolutely no reason. Just prefer it that way :)

---
In reply to: [212401411](https://github.com/Microsoft/ChakraCore/pull/5623#discussion_r212401411) [](ancestors = 212401411)

**Comment by duongnhn** [comment]
will do

---
In reply to: [415512714](https://github.com/Microsoft/ChakraCore/pull/5623#issuecomment-415512714) [](ancestors = 415512714)

File: bin/ch/WScriptJsrt.cpp (hunk 2 above)
Comment (duongnhn) [reply bin/ch/WScriptJsrt.cpp]: make sense. will do.

---
In reply to: [212402959](https://github.com/Microsoft/ChakraCore/pull/5623#discussion_r212402959) [](ancestors = 212402959)

**Comment by jackhorton** [comment]
@dotnet-bot test OSX static_osx_osx_debug please

**Pull request merged**